subsys/event_manager/event_manager.c

```diff
@@ -175,7 +160,9 @@ static void event_processor_fn(struct k_work *work)
 
 		const struct event_type *et = eh->type_id;
 
-		event_manager_trace_event_execution(eh, true);
+		STRUCT_SECTION_FOREACH(event_preprocess_hook, h) {
+			h->hook(eh);
```

Comment:
Im not sure if its possible to add `BUILD_ASSERT` when using `REGISTER` macros, but if no we should
```
ASSERT_NO_MSG(h != NULL);
ASSERT_NO_MSG(h->hook != NULL);
```
Similarly as for `el->notification` below.

Or does `STRUCT_SECTION_FOREACH` does that for us?

Reply:
There is no way that h would equal NULL. The hook could... theoretically. But you have to really, really overuse the API to be able to do that. I am not even sure if it would compile then.

Reply:
We are really not using it when registering events? Do we really wish to protect the user from this kind of usage:
```
EVENT_HOOK_PREPROCESS_REGISTER(NULL);
```
??

Reply:
I just made same assumption as it was done for event listener functions. For me it also seems to silly to pass NULL as a function handler but you never know :P

Reply:
I had added build assert - but still think it is an overkill.